.github/workflows/CI.yml

```
@@ -71,7 +71,7 @@ jobs:
       - uses: codecov/codecov-action@v4
         with:
           files: lcov.info
-
+          token: ${{ secrets.CODECOV_TOKEN }}
```

This can be a good way to get immediate feedback when Codecov isn't working.

```suggestion
          token: ${{ secrets.CODECOV_TOKEN }}
                  fail_ci_if_error: true
```